runtime-transaction/src/signature_details.rs

```diff
@@ -0,0 +1,175 @@
+// static account keys has max
+use {
+    agave_transaction_view::static_account_keys_frame::MAX_STATIC_ACCOUNTS_PER_PACKET as FILTER_SIZE,
+    solana_sdk::{message::TransactionSignatureDetails, pubkey::Pubkey},
+    solana_svm_transaction::instruction::SVMInstruction,
+};
+
+/// Get transaction signature details.
+pub fn get_signature_details<'a>(
+    num_transaction_signatures: u64,
+    instructions: impl Iterator<Item = (&'a Pubkey, SVMInstruction<'a>)>,
+) -> TransactionSignatureDetails {
+    let mut filter = SignatureDetailsFilter::new();
+
+    // Wrapping arithmetic is safe below because the maximum number of signatures
+    // per instruction is 255, and the maximum number of instructions per transaction
+    // is low enough that the sum of all signatures will not overflow a u64.
```

Comment:
Results below if we use `saturating_add` instead:

<details>
<summary>Benchmark Results</summary>

```
bench_get_signature_details_empty/0 instructions
                        time:   [247.50 ps 247.66 ps 247.86 ps]
                        thrpt:  [4.0345 Gelem/s 4.0379 Gelem/s 4.0404 Gelem/s]
                 change:
                        time:   [-0.4477% -0.0591% +0.2935%] (p = 0.76 > 0.05)
                        thrpt:  [-0.2926% +0.0592% +0.4498%]
                        No change in performance detected.

bench_get_signature_details_no_sigs_unique/4 instructions
                        time:   [6.6299 ns 6.6350 ns 6.6403 ns]
                        thrpt:  [150.59 Melem/s 150.71 Melem/s 150.83 Melem/s]
                 change:
                        time:   [-0.4119% -0.1811% +0.0343%] (p = 0.12 > 0.05)
                        thrpt:  [-0.0343% +0.1814% +0.4136%]
                        No change in performance detected.

bench_get_signature_details_no_sigs_unique/32 instructions
                        time:   [40.811 ns 40.874 ns 40.947 ns]
                        thrpt:  [24.422 Melem/s 24.466 Melem/s 24.503 Melem/s]
                 change:
                        time:   [+0.2210% +0.6105% +1.0064%] (p = 0.00 < 0.05)
                        thrpt:  [-0.9964% -0.6068% -0.2205%]
                        Change within noise threshold.

bench_get_signature_details_packed_sigs/4 instructions
                        time:   [6.8985 ns 6.9025 ns 6.9062 ns]
                        thrpt:  [144.80 Melem/s 144.88 Melem/s 144.96 Melem/s]
                 change:
                        time:   [+7.5403% +7.8383% +8.1052%] (p = 0.00 < 0.05)
                        thrpt:  [-7.4975% -7.2686% -7.0116%]
                        Performance has regressed.

bench_get_signature_details_packed_sigs/64 instructions
                        time:   [69.559 ns 69.700 ns 69.893 ns]
                        thrpt:  [14.308 Melem/s 14.347 Melem/s 14.376 Melem/s]
                 change:
                        time:   [+5.0905% +5.4515% +5.8803%] (p = 0.00 < 0.05)
                        thrpt:  [-5.5537% -5.1697% -4.8440%]
                        Performance has regressed.

bench_get_signature_details_mixed_sigs/4 instructions
                        time:   [7.3652 ns 7.3720 ns 7.3804 ns]
                        thrpt:  [135.49 Melem/s 135.65 Melem/s 135.77 Melem/s]
                 change:
                        time:   [+3.4887% +3.7886% +4.1034%] (p = 0.00 < 0.05)
                        thrpt:  [-3.9417% -3.6503% -3.3711%]
                        Performance has regressed.

bench_get_signature_details_mixed_sigs/64 instructions
                        time:   [53.674 ns 53.700 ns 53.727 ns]
                        thrpt:  [18.613 Melem/s 18.622 Melem/s 18.631 Melem/s]
                 change:
                        time:   [+0.9589% +1.3386% +1.7877%] (p = 0.00 < 0.05)
                        thrpt:  [-1.7563% -1.3209% -0.9498%]
                        Change within noise threshold.

```

</details>

Roughly 7% worse perf in worst case.
In **either** case the math is wrong but wrapping is probably "more" wrong.
I think we never need worry about such limits though, with the current **instruction** format (if that changed we'd need to re-write this anyway), we cannot reasonably hit limits regardless of our transaction size.
To hit u64 limits we'd need 72340172838076673 (72 QUADRILLION) instructions...I just don't see it happening 😆 
To even come close we'd also need to change transaction format, since the number of instructions is currently u16::MAX if we ignore packet size.

Reply:
thanks for benching data. What am I uncomfortable about it is there is a implicitly implied limit, deeply buried in the code. However, your reasoning is valid that this limit will never practically hit. My concern is more theoretical than practical. Should be OK to take that 7% perf gain :)